core/src/main/java/org/lflang/target/Target.java

```diff
@@ -546,4 +580,84 @@ public static Target fromDecl(TargetDecl targetDecl) {
     return Target.forName(name)
         .orElseThrow(() -> new RuntimeException("Invalid target name '" + name + "'"));
   }
+
+  public void initialize(TargetConfig config) {
+    switch (this) {
+      case C, CCPP -> config.register(
+          AuthProperty.INSTANCE,
+          BuildCommandsProperty.INSTANCE,
+          BuildTypeProperty.INSTANCE,
+          ClockSyncModeProperty.INSTANCE,
+          ClockSyncOptionsProperty.INSTANCE,
+          CmakeIncludeProperty.INSTANCE,
+          CompileDefinitionsProperty.INSTANCE,
+          CompilerFlagsProperty.INSTANCE,
+          CompilerProperty.INSTANCE,
+          CoordinationOptionsProperty.INSTANCE,
+          CoordinationProperty.INSTANCE,
+          DockerProperty.INSTANCE,
+          FilesProperty.INSTANCE,
+          HierarchicalBinProperty.INSTANCE,
+          KeepaliveProperty.INSTANCE,
+          PlatformProperty.INSTANCE,
+          ProtobufsProperty.INSTANCE,
+          SchedulerProperty.INSTANCE,
+          ThreadingProperty.INSTANCE,
+          TracingProperty.INSTANCE,
+          VerifyProperty.INSTANCE,
+          WorkersProperty.INSTANCE);
+      case CPP -> config.register(
+          BuildTypeProperty.INSTANCE,
+          CmakeIncludeProperty.INSTANCE,
+          CompilerProperty.INSTANCE,
+          ExportDependencyGraphProperty.INSTANCE,
+          ExportToYamlProperty.INSTANCE,
+          ExternalRuntimePathProperty.INSTANCE,
+          NoRuntimeValidationProperty.INSTANCE,
+          PrintStatisticsProperty.INSTANCE,
+          Ros2DependenciesProperty.INSTANCE,
+          Ros2Property.INSTANCE,
+          RuntimeVersionProperty.INSTANCE,
+          TracingProperty.INSTANCE,
```

Comment:
We should distinguish the C tracing property (which can be configured) from the C++ property (which is a plain boolean).

Reply:
I don't suggest we do this because it will become too complicated. What I will do is add a validation check to catch inappropriate use.

Reply:
I do consider this refactoring a success based on how easy it was to implement this check. See [8018f7f](https://github.com/lf-lang/lingua-franca/pull/2008/commits/8018f7f2744f4522878056fc6317548a6ef08b27).